cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -47,6 +47,7 @@ type AutoscalingGceClient interface {
 	// reading resources
 	FetchMachineType(zone, machineType string) (*gce.MachineType, error)
 	FetchMachineTypes(zone string) ([]*gce.MachineType, error)
+	ListMigs(zone string) ([]*gce.InstanceGroupManager, error)
```

Comment:
1) I don't like returning *gce.InstanceGroupManager here. The reason this object (autoscalingGceClientV1) exists is to hide as much as possible of stuff from GCE API, so that the rest of the code doesn't need to know (or care) what kind, and how many requests are made.
I realize this is already not the case for MachineType and InstanceTemplate (because processing templates is messy), but here we only need target sizes. 
2) Following the pattern from autodiscovery (FetchMigsWithName) and getting only MIGs we're interested in, I'd prefer if this method accepted []GceRef of configured MIGs and filtered them inside. 
3) Lastly, please follow the 'Fetch' naming convention in this file. It may not be perfect but at least it's consistent;)

Reply:
> 1. I don't like returning *gce.InstanceGroupManager here. The reason this object (autoscalingGceClientV1) exists is to hide as much as possible of stuff from GCE API, so that the rest of the code doesn't need to know (or care) what kind, and how many requests are made.
> I realize this is already not the case for MachineType and InstanceTemplate (because processing templates is messy), but here we only need target sizes.

I will pass on this one. As you noticed we are not doing it for other entitiies anyway. And it does not seem that creating a wrapper struct around targetSize will do us any good right now.

> 2. Following the pattern from autodiscovery (FetchMigsWithName) and getting only MIGs we're interested in, I'd prefer if this method accepted []GceRef of configured MIGs and filtered them inside.

I do not see any benefit of that (given you cannot filter out migs on the server side) . And such interface would be more cumbersome to use.  I will skip it for now.

> 3. Lastly, please follow the 'Fetch' naming convention in this file. It may not be perfect but at least it's consistent;)

Done

Reply:
> I do not see any benefit of that (given you cannot filter out migs on the server side) 

I take that back. Actually you can filter out using "ORs" and multiple condition on mig name. I will ask MIGs team what is the expected performance and limitations of that, and possibly make the change so this method takes a list of GCERefs.

Reply:
> I take that back. Actually you can filter out using "ORs" and multiple condition on mig name. I will ask MIGs team what is the expected performance and limitations of that, and possibly make the change so this method takes a list of GCERefs.

I am doing some more testing related to filtering - but based on conversation it seems it would not help us much (in terms of performance) to pass list of refs as arg. Let's merge this one and iterate with more data.